src/plugins/data_views/common/data_views/data_views.ts

```diff
@@ -572,7 +572,9 @@ export class DataViewsService {
       ...options,
       pattern: indexPattern.title as string,
       allowHidden:
-        (indexPattern as DataViewSpec).allowHidden || (indexPattern as DataView)?.getAllowHidden(),
+        (indexPattern as DataViewSpec).allowHidden == null
```

Comment:
I'm pretty sure this resolves to `false` when `indexPattern.allowHidden` is true. 

It would also be nice to have a test for this, as best I can tell it should be simple.

Reply:
Yes, you are right, overlook on my side.

Thanks for catching it, I also added tests you suggested

Reply:
@mattkime 

yes, you are right.
Thanks for catching it

I also added tests as you suggested in your PR: https://github.com/elastic/kibana/pull/180800/commits/06dad88845440d9279fc1cf156bc2779a5699029